.github/workflows/subxt-diff.yml

```diff
@@ -2,20 +2,42 @@ name: Subxt Diff
 on:
   workflow_dispatch:
     inputs:
+      last_release_sha8:
+        description: The sha8 of the last release, with which to compare local to.
+        required: true
       spec_version:
         description: runtime spec version (ex. 1601)
         required: true
 
 jobs:
+  build:
+    runs-on:
+      labels: bare-metal
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Local build new Node
+        uses: ./.github/workflow-templates/cargo-build
+      - name: Upload Node
+        uses: actions/upload-artifact@v4
+        with:
+          name: moonbeam
+          path: build
+
   local-diff:
-    runs-on: ubuntu-latest
+    needs: build
+    runs-on: bare-metal
+    strategy:
+      matrix:
+        runtime: [moonbeam, moonbase, moonriver]
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
-      - name: Use pnpm
-        uses: pnpm/action-setup@v2
+        uses: actions/checkout@v4
+      - name: Download Node
+        uses: actions/download-artifact@v4
         with:
-          version: 8.6.12
+          name: moonbeam
+          path: build
       - name: Install Rust toolchain
```

Comment:
Please use "rustup show" to install rust the same way as other workflows, I think that this step is the cause that break the CI